barman/compression.py

```diff
@@ -565,6 +573,36 @@ def validate(self, pg_server_version, remote_status):
         return issues
 
 
+class LZ4PgBaseBackupCompressionOption(PgBaseBackupCompressionOption):
+    def validate(self, pg_server_version, remote_status):
+        """
+        Validate lz4-specific options.
+
+        :param pg_server_version int: the server for which the
+          compression options should be validated.
+        :param dict remote_status: the status of the pg_basebackup command
+        :return List: List of Issues (str) or empty list
+        """
+        issues = super(LZ4PgBaseBackupCompressionOption, self).validate(
+            pg_server_version, remote_status
+        )
+        # "backup_location = server" requires pg_basebackup >= 15
+        if remote_status["pg_basebackup_version"] < Version("15"):
+            issues.append(
+                "backup_compression = %s requires "
+                "pg_basebackup 15 or greater" % self.config.type
+            )
+
+        if self.config.level is not None and (
+            self.config.level < 0 or self.config.level > 9
```

Comment:
I think the range for the `lz4` cli is 1 to 12.

Reply:
I have found several limits according to where I search.

lz4 -H on ubuntu sais:
```shell
Compression levels : 
There are technically 2 accessible compression levels.
-0 ... -2 => Fast compression
-3 ... -9 => High compression
```

another help [page](https://www.mankier.com/1/lz4) does say 1 to 12

Maybe I did not instal the proper binary or there is missing information on the help ?

Reply:
Ah interesting. I tried the liblz4-tool package on ubuntu 18.04 which lists levels 0 to 9 though if I use a higher value such as `-12` it still works (though the file was not compressed any more than with `-9`).

With ubuntu 20.04 the lz4 command installed by the `lz4` package implies `-12`.

Maybe we should just validate it's non-negative and leave it at that?

Reply:
I tried with a high value and got  following mesage:
```
pg_basebackup: error: invalid compression specification: compression algorithm "lz4" expects a compression level between 1 and 12
```
I guess that settles it :)